examples/hsgp.py

```diff
@@ -221,7 +221,7 @@ def approx_se_ncp(x, alpha, length, L, M):
     """
     phi = eigenfunctions(x, L, M)
     spd = jnp.sqrt(diag_spectral_density(alpha, length, L, M))
-    with plate("basis", M):
+    with plate(f"basis_{M}", M):
```

Comment:
This `basic` plate has different sizes in different scopes. So we need to add `M` here to fix the issue. This raises some questions around what would be a reasonable behavior of `scope`.